wordpress-mobile/WordPress-iOS
# Gutenberg/add gallery processors

Fixes #https://github.com/wordpress-mobile/gutenberg-mobile/issues/1629

This PR add a new processor that is specific for Gallery blocks.
It handles 3 possible scenarios for galleries, that depend of the link to value:
 - none: there is no `a` element is in the gallery and we just have img
 - media:  we have a `a` element and it points to the full url of the img
 - attachment: we have a 'a' element and it should point to the attachment page for media. I noticed that the moment the gallery block is not inserting the `a` element because the `link` field for media is not defined. Fixing this should be done on the main gallery block PR because we will need to start to send the link information for image/media to the JS/React side. This will need to change to the bridge code and in the main apps to retrieve that information.

To test:
 - Create a new post using Guteberg
 - Add a gallery block with some images
 - While the images are still uploading press publish
 - Wait outside the editor for the post to be published
 - Open the post again
 - Check that the gallery block is working
 - Switch to HTML mode and see if the link and id are pointing to the server/remote values.

PR submission checklist:
